lib/src/media/stream/client.dart

```diff
@@ -1,36 +1,45 @@
 import 'dart:async';
 
 import 'package:flutter_webrtc/flutter_webrtc.dart';
+import 'package:logger/logger.dart';
 
 import '../../gen/proto/stream/v1/stream.pbgrpc.dart';
 import '../../rpc/web_rtc/web_rtc_client.dart';
 
+Logger _logger = Logger();
+
 class StreamManager {
   final Map<String, MediaStream> _streams = {};
   final Map<String, StreamClient> _clients = {};
   WebRtcClientChannel _channel;
-  final StreamServiceClient _client;
   // ignore: cancel_subscriptions
   StreamSubscription? _errorHandler;
 
   static final Finalizer<StreamSubscription?> _finalizer = Finalizer((p0) {
     p0?.cancel();
   });
 
-  StreamManager(this._channel) : _client = StreamServiceClient(_channel) {
+  StreamManager(this._channel) {
     _finalizer.attach(this, _errorHandler);
     channel = _channel;
   }
 
+  StreamServiceClient get _client {
+    return StreamServiceClient(_channel);
+  }
+
   set channel(WebRtcClientChannel channel) {
+    _errorHandler?.cancel();
     _channel = channel;
     _channel.rtcPeerConnection.onTrack = (event) {
+      _errorHandler?.cancel(); // Cancel the error handler -- clearly we're connected if we're receiving this event
```

Comment:
(q) why are we doing this twice?

Reply:
The first time happens whenever we change the channel.

This one is on a callback -- so it's possible it gets called out of order/after the error handler exists again.

Reply:
The first one happens whenever the channel changes. This one is happening on a callback which could be called out of order/delayed (after  the `_errorHandler` exists again). The one that happens a bit further down is in another callback, where we want to cancel if the returned state is not an error state.